src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,351 @@
+    }
+    else{
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Collider 'M' in reaction '{}' cannot contain both eps and eig0. Any"
+            " additional colliders must also make same choice as that of M.", eqn);
+    }
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
+    // Start at 1 because index 0 is for "M"
+    for (size_t i = 1; i < colliders.size(); i++){
+        if (!colliders[i].hasKey("name")) {
+                throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                    "The collider located at index '{}' in 'colliders' of reaction '{}'"
+                    " has no 'name' defined.", std::to_string(i), eqn);
+        }
+        auto nm = colliders[i]["name"].asString();
+        if (!colliders[i].hasKey("eps") && !colliders[0].hasKey("eig0")) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "Collider '{}' in reaction '{}' is missing an 'eps' or 'eig0' key.",
+                nm, eqn);
+        }
+        if (!colliders[i].hasKey(eig_eps_key)) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "All collision efficiencies in reaction '{}' must be defined uniformly"
+                " as either eig0 or eps. No mixing and matching is allowed.", eqn);
+        }
+        // Save data to colliderInfo, which will make it accessible by getParameters
+        colliderInfo.insert({colliders[i]["name"].as<string>(), colliders[i]});
+        colliderNames.push_back(colliders[i]["name"].as<string>());
+        ArrheniusRate epsObj_i;
+        // eig0 and eps are ONLY interchangeable due to the requirement that eps_M have
+        // parameters {A: 1, b: 0, Ea: 0}
+        params["A"] = colliders[i][eig_eps_key]["A"].asDouble() /
+            colliders[0][eig_eps_key]["A"].asDouble();
+        params["b"] = colliders[i][eig_eps_key]["b"].asDouble() -
+            colliders[0][eig_eps_key]["b"].asDouble();
+        params["Ea"] = colliders[i][eig_eps_key]["Ea"].asDouble() -
+            colliders[0][eig_eps_key]["Ea"].asDouble();
+
+        if (params["A"].asDouble() < 0) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "Invalid eig0 or eps entry for one or more colliders.");
+        }
+
+        epsObj_i = ArrheniusRate(AnyValue(params), colliders[i].units(), eps_units);
+        if (colliders[i].hasKey("rate-constants")) {
+            rateObjs.push_back(PlogRate(colliders[i], rate_units));
+            dataObjs.push_back(PlogData());
+            epsObjs1.push_back(epsObj_i);
+            epsObjs2.push_back(epsObj_i);
+        }
+        else if (colliders[i].hasKey("Troe")) {
+            // Value of "type" is unimportant; just needed to make falloff.cpp run
+            colliders[i]["type"];
```

Comment:
Rather than relying on the fact that this isn't explicitly checked:
```suggestion
            colliders[i]["type"] = "falloff";
```